### PR TITLE
Flag Advanced Settings

### PR DESCRIPTION
**Pull request by foodprocessor**
Add const flag which customers can set true to remove advanced settings from plugin UI.
This flag will always stay true in this repo and is for customer use only.
When the flag is true, `kDefaultEndpoint` will be used as the endpoint. This is meant to allow the customer to set a default endpoint without having to think about what Cloudfuse's default endpoint is.

### What type of Pull Request is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

### Checklist

- [x] Tested locally
- [ ] Added new dependencies
- [ ] Updated documentation
- [ ] Added tests

### Related Issues

- Related Issue #
- Closes #

### CODE REVIEW

File: src/plugin/settings/engine.cpp [hunk 1]
```diff
@@ -480,9 +489,7 @@ nx::sdk::Error Engine::spawnMount()
 #endif
     if (mountRet.errCode != 0)
     {
-        std::string errorMessage = "Unable to launch mount with error: " + mountRet.output;
```
Comment (jfantinhardesty) [inline src/plugin/settings/engine.cpp:483]: Is there a reason to not log these messages using NX_PRINT?

File: src/plugin/settings/engine.cpp [hunk 2]
```diff
@@ -525,6 +530,15 @@ void Engine::doGetSettingsOnActiveSettingChange(Result<const IActiveSettingChang
 
     std::map<std::string, std::string> values = toStdMap(shareToPtr(activeSettingChangedAction->settingsValues()));
 
+    // if (!updateModel(&model, &values, {settingId}))
```
Comment (jfantinhardesty) [inline src/plugin/settings/engine.cpp:533]: Do you want this commented out?

**Review by jfantinhardesty — APPROVED** [review]

File: src/plugin/settings/engine.cpp [hunk 3]
```diff
@@ -22,6 +22,7 @@
 
 #include "cloudfuse_helper.h"
 
+// #define NX_PRINT_PREFIX (this->logUtils.printPrefix)
```
[the remark this answers is not in the document]
Comment (foodprocessor) [reply src/plugin/settings/engine.cpp:25]: I removed this line in a previous PR, and afterwards realized that this gives us a nice prefix to our NX_PRINT output, with a timestamp and some other metadata.
So, I want to bring this back, but it doesn't work inside helper functions like the `enableLogging()` function.
So, I'm thinking to either make the helper functions in to class members, or to figure out how to get that nice prefix without using `this->`.
Any suggestions on the way forward with that are welcome.
For now, I thought I'd leave it commented as a TODO for myself. Maybe I'll actually make it an explicit TODO...

File: src/plugin/settings/engine.cpp (hunk 1 above)
Comment (foodprocessor) [reply src/plugin/settings/engine.cpp:483]: It turns out they're already logged after being returned. I only realized this when the mount actually failed recently.

File: src/plugin/settings/engine.cpp (hunk 2 above)
Comment (foodprocessor) [reply src/plugin/settings/engine.cpp:534]: Woops! There's that commented block, back from the dead! This was from a bad rebase. I'll drop the block (again!).

**Review by jfantinhardesty — APPROVED** [review]

**Pull request merged**